lib/password.c

```diff
@@ -0,0 +1,237 @@
+#define BITLBEE_CORE
+
+#include <gcrypt.h>
+#include "bitlbee.h"
+#include "base64.h"
+#include "arc.h"
+#include "md5.h"
+
+
+/* pbkdf2-hashed password. We store an algoritm identifier ("1"), the salt
+ * used, the number of rounds and the final hash */
+size_t password_hash(const char *password, char **hash)
+{
+	unsigned char rawhash[64], salt[8];
+	char *b64hash, *b64salt;
+	int saltlen = 8, rounds = 10000, keylen = 64;
+
+	random_bytes(salt, saltlen);
+	gcry_kdf_derive(password, strlen(password), GCRY_KDF_PBKDF2, GCRY_MD_SHA1,
```

Comment:
From what I've read, doing sha1 is standard here. I know sha256 is stronger but the crypto community seemed ambivalent about it. I don't care much either way, so I'll change it.